### ISSUE TITLE
Hydra builds Julia for the wrong architecture

### ISSUE THREAD

**Issue by ttuegel** [NixOS/nixpkgs#9921]
Julia *claims* that the default options result in a generic build, but the latest Hydra build won't run on my machine. It dies with the error
```
Target architecture mismatch. Please delete or regenerate sys.{so,dll,dylib}.
```

The problem begins with 02fc4551f5d1eeb657cddea4658eb4212518311c, which lets Julia automatically detect the architecture instead of specifying a generic one. I think the Julia documentation really means, the default *binary* is a generic build.

@abbradar I am reverting this commit and removing `i686-linux` from the build platforms. To properly fix the build error on i686, we need to figure out how to pass the correct architecture to Julia's LLVM build. Or, better yet, use the existing LLVM 3.3 build + Julia's LLVM 3.3 patch to build a just-for-Julia LLVM within Nixpkgs.

**Comment by ttuegel** [comment]
The exact reason it is necessary to set `MARCH` is described [here](https://github.com/JuliaLang/julia/blob/master/DISTRIBUTING.md). This is all fixed as of 78634c7a547ad634976fd24a44eb477377de454a.

**Comment by abbradar** [comment]
Thank you very much!
-- 
Nikolay.
